modules/sdl2/sdl.c

```diff
@@ -30,6 +30,25 @@ struct vidisp_st {
 static struct vidisp *vid;
 
 
+static enum vidfmt match_fmt(u_int32_t fmt)
+{
+	switch (fmt) {
+	case VID_FMT_YUV420P:	return SDL_PIXELFORMAT_IYUV;
+	case VID_FMT_YUYV422:	return SDL_PIXELFORMAT_YUY2;
+	case VID_FMT_UYVY422:	return SDL_PIXELFORMAT_UYVY;
+	case VID_FMT_RGB32:	return SDL_PIXELFORMAT_RGBA8888;
+	case VID_FMT_ARGB:	return SDL_PIXELFORMAT_ARGB8888;
+	case VID_FMT_RGB565:	return SDL_PIXELFORMAT_RGB565;
+	case VID_FMT_RGB555:	return SDL_PIXELFORMAT_RGB555;
+	case VID_FMT_NV12:	return SDL_PIXELFORMAT_NV12;
+	case VID_FMT_NV21:	return SDL_PIXELFORMAT_NV21;
+	case VID_FMT_YUV444P:
+	case VID_FMT_N:
+	default:		return SDL_PIXELFORMAT_UNKNOWN;
```

Comment:
have you tested that all these pixel formats are working?

we dont have to add all formats now, I think it is better to just add a couple of
formats that is tested and working, such as YUV420P and NV12.